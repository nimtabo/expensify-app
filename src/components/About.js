import React from 'react';

const AboutPage = () => (
  <div>
    <div className="page-header">
      <div className="content-container">
        <h2 className="page-header__title">Developer: Nicholas Mtabo</h2>
        <p>nicholasmtabo@yahoo.com</p>
        <p>+254 704 643 061</p>
      </div>
    </div>

    <div className="content-container">
      <img className="image__item" src="/images/about/react.png" alt="React" title="React" />
      <img className="image__item" src="/images/about/redux.png" alt="Redux" title="Redux" />
      <img className="image__item" src="/images/about/node.png" alt="Node.js" title="Nodejs" />
      <img className="image__item" src="/images/about/sasss.png" alt="Sass" title="Sass" />
      <img className="image__item" src="/images/about/firebase.png" alt="firebase" title="firebase" />
      <img className="image__item" src="/images/about/webpack.png" alt="webpack" title="webpack" />
      <img className="image__item" src="/images/about/jestenzyme.jpg" alt="jest-enzyme" title="jest-enzyme" />
    </div>
  </div>
);

export default AboutPage;