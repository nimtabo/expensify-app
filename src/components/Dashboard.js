import React from 'react';
import ExpenseList from './Expenselist';
import ExpenseListFilters from './ExpenseListFilters';


const ExpenseDashboardPage = () => (
  <div>
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;