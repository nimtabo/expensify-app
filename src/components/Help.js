import React from 'react';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';

const HelpPage = () => (
  <div>
    This is from Help page
  </div>
);

export default HelpPage;