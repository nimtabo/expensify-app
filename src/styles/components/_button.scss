.button {
  background-color: $blue;
  border: none;
  border-radius: 10px;
  color: azure;
  display: inline-block;
  font-weight: 300;
  font-size: $font-size-large;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

// button link modifier
.button--link {
  background: none;
}

.button--secondary {
  background: #888;
}