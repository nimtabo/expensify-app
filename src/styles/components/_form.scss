.form {
  display: flex;
  flex-direction: column;
  > * {
    margin-bottom: $m-size;
  }
}

.form__error {
  margin: 0 0 $m-size 0;
  color: darkgreen;
  font-style: italic;
  font-weight: 700;
  text-align: center;
}