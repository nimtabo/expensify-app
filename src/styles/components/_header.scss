.header {
  background: $dark-blue;
}

.header__about {
  color: white;
  font-size: $font-size-large;
  text-decoration: none;
}

.header__content {
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding: $s-size 0;
}

.header__title {
  color: white;
  text-decoration: none;
  h1 {
    margin: 0;
  }
}