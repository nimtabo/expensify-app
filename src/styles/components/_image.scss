.image__item {
  width: 75%;
  float: left;
  margin: 6.25%;
  @media (min-width: $desktop-breakpoint) {
    width: 30%;
    float: left;
    margin: 1.66%;
 }
}