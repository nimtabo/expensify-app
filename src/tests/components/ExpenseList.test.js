import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/Expenselist';
import expenses from '../seed/expenses';

test('should render ExpenseList with expenses', () => {
  const wrapper = shallow(<ExpenseList expenses={expenses}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseList with empty message', () => {
  const wrapper = shallow(<ExpenseList expenses={[]}/>);
  expect(wrapper).toMatchSnapshot();
});