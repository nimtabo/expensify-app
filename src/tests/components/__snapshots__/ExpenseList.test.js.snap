// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with empty message 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <p>
    No expenses
  </p>
</div>
`;

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <ExpenseListItem
    amount={4000}
    createdAt={0}
    description="cable"
    id="1"
    key="1"
    note="cable bill"
  />
  <ExpenseListItem
    amount={7000}
    createdAt={-345600000}
    description="water"
    id="2"
    key="2"
    note="water bill"
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="power"
    id="3"
    key="3"
    note="electric bill"
  />
</div>
`;
